### DISCUSSION TITLE
nanoCUL using Arduino Nano Every

### DISCUSSION THREAD

**Discussion by DecksLabs** [wmbusmeters/wmbusmeters#582] [Show and tell]
Hi,

I've ported nanoCUL to Arduino Nano Every.

I wasn't really happy with the price of commercial wireless MBUS dongles or using RTL-SDR due to it's power consumption and load on CPU. As the original nanoCUL is not able to receive longer telegrams I've ported it to the newer [Arduino Nano Every](https://github.com/DecksLabs/every-culfw/). It has three times more the RAM which made it possible to get five times longer UART buffer out of it, though I'm not sure if it is enough to support every mbus telegram there is but at least I tried. Although I've stripped the project of... nearly everthing, I think there is still some room for improvement.

I've also written [some documentation](https://github.com/DecksLabs/every-culfw/tree/main/docs) on making this stick yourself and flashing Nano Every with it.

Enjoy.

**Comment by trsqr** [comment]
Very cool, thanks! Does this work as nanoCUL from wmbusmeters point of view?

**Comment by DecksLabs** [comment]
Yes, it's totally transparent.

niedz., 20 lis 2022, 09:41 użytkownik Olli Salonen ***@***.***>
napisał:
